src/pages/[platform]/build-a-backend/graphqlapi/connect-api-to-existing-database/index.mdx

```diff
@@ -0,0 +1,834 @@
+
+```json
+{
+  "data": {
+    "publishPosts": "{\"fieldCount\":0,\"affectedRows\":7,\"insertId\":0,\"info\":\"Rows matched: 7  Changed: 7  Warnings: 0\",\"serverStatus\":34,\"warningStatus\":0,\"changedRows\":7}"
+  }
+}
+```
+
+However, you might want to return the actual row data instead.
+
+<BlockSwitcher>
+<Block name="MySQL">
+
+In MySQL, you can create and call a stored procedure that performs both an UPDATE statement and SELECT query to return a single post.
+
+Create a stored procedure by running the following SQL statement in your MySQL database:
+
+```sql
+CREATE PROCEDURE publish_post (IN postId VARCHAR(255))
+
+BEGIN
+UPDATE posts SET published = 1 WHERE id = postId;
+
+SELECT * FROM posts WHERE id = postId LIMIT 1;
+END
+```
+
+Call the stored procedure from the custom mutation:
+
+```graphql
+type Mutation {
+  publishPostById(id: String!): [Post]
+    @sql(statement: "CALL publish_post(:id);")
+    @auth(rules: [{ allow: public }])
+}
+```
+
+</Block>
+<Block name="PostgreSQL">
+
+In PostgreSQL, you can add a `RETURNING` clause to an `INSERT`, `UPDATE`, or `DELETE` statement and get the actual modified row data.
+
+Example:
+
+```graphql
+type Mutation {
+  publishPostById(id: String!): [Post]
+    @sql(statement: "UPDATE posts SET price = :id RETURNING *;")
+    @auth(rules: [{ allow: public }])
+}
+```
+
+</Block>
+</BlockSwitcher>
+
+<Callout info>
+  The return type for custom queries and mutations expecting row data must
+  be an array of the corresponding model.
+</Callout>
+
+
+## Apply authorization rules
+
+The `@auth` directive can be used to restrict access to data and operations by specifying authorization rules. It allows granular access control over the GraphQL API based on the user's identity and attributes. You can for example, limit a query or mutation to only logged-in users via an `@auth(rules: [{ allow: private }])` rule or limit access to only users of the "Admin" group via an `@auth(rules: [{ allow: groups, groups: ["Admin"] }])` rule.
+
+{/* All model-level authorization rules are supported for Amplify GraphQL schemas generated from MySQL and PostgreSQL databases.
+
+<Callout warning>**Known limitation:** Field level auth rules are not supported.</Callout>
+
+In the example below, public users authorized via API Key are granted unrestricted access to all posts.
+
+Add the following auth rule to the `Blog` model within the `schema.sql.graphql` file:
+
+```graphql
+type Blog @model @refersTo(name: "blogs") @auth(rules: [{ allow: public }]) {
+  id: String! @primaryKey
+  title: String!
+}
+``` */}
+
+{/* In a real world scenario, you can instead define auth rules that only allow public users to read posts, and authenticated users the ability to update or delete their posts. */}
+
+For more information on each rule please refer to our documentation on [Authorization rules](/[platform]/build-a-backend/graphqlapi/customize-authorization-rules/).
+
+## Deploy your API
+
+<BlockSwitcher>
```

Comment:
@chrisbonifacio Not sure if this was left over edit but not sure why this is in a blockswitcher with only one entry.